xdem/coreg/biascorr.py

```diff
@@ -230,11 +251,24 @@ def _fit_biascorr(  # type: ignore
                     "with function {}.".format(", ".join(list(bias_vars.keys())), self._meta["fit_func"].__name__)
                 )
 
+            if isinstance(diff, np.ndarray):
```

Comment:
Here is the data subsampling that can be moved out of the if loop "Option 1: Run fit...", then `xdata`/`ydata`/etc can be passed to every option similarly :slightly_smiling_face:

Reply:
I guess [this](https://github.com/GlacioHack/xdem/pull/525/commits/1c57a389b43bb252eedbf7f92f4164d6b9e443cc) is what you meant? :slightly_smiling_face:

Reply:
which of the unit test would you expect to pass if passed an xarray instead of the normal input?

Reply:
Yes exactly!
I guess for tests with only `fit()`, everything should pass exactly the same with just the Xarray input.
For tests that also have `apply()`, you'd have to write a check on the input type and call `.compute()` to be able to run the final checks on the output arrays.

Reply:
So I think you should be able to pass everything by just adding an Xarray input in the current list: https://github.com/ameliefroessl/xdem/blob/1c57a389b43bb252eedbf7f92f4164d6b9e443cc/tests/test_coreg/test_biascorr.py#L97
And adding a `compute()` only for tests using the output of `apply()` (the ones just calling it don't need to), like here: https://github.com/ameliefroessl/xdem/blob/1c57a389b43bb252eedbf7f92f4164d6b9e443cc/tests/test_coreg/test_biascorr.py#L520, and other synthetic tests further below.